Wrappers/Python/cil/optimisation/operators/Operator.py

```diff
@@ -187,31 +187,34 @@ def dot_test(operator, domain_init=None, range_init=None, verbose=False, **kwarg
         :param domain_init: optional initialisation container in the operator domain 
         :returns: boolean, True if the test is passed.
         :param decimal: desired precision
-        :type decimal: int, optional, default 4
+        :type decimal: int, optional, default 4        
         '''
-        seed = kwargs.get('seed',None)
-        if range_init is None:
-            y = operator.range_geometry().allocate('random', seed=seed)
-        else:
-            y = range_init
-        if domain_init is None:
-            x = operator.domain_geometry().allocate('random',seed=seed)
-        else:
-            x = domain_init
+
+        lst = [1,2,3]
+        for i in lst:
+        # seed = kwargs.get('seed',None)
+            if range_init is None:
+                y = operator.range_geometry().allocate('random', seed = i)
+            else:
+                y = range_init
+            if domain_init is None:
+                x = operator.domain_geometry().allocate('random',seed = i)
+            else:
+                x = domain_init
             
-        fx = operator.direct(x)
-        by = operator.adjoint(y)
-        a = fx.dot(y)
-        b = by.dot(x).conjugate()
-        if verbose:
-            print ('Left hand side  {}, \nRight hand side {}'.format(a, b))
-            print ("decimal ", kwargs.get('decimal', 4))
-        try:
-            numpy.testing.assert_almost_equal(abs((a-b)/a), 0., decimal=kwargs.get('decimal',4))
-            return True
-        except AssertionError as ae:
-            print (ae)
-            return False
+            fx = operator.direct(x)
+            by = operator.adjoint(y)
+            a = fx.dot(y)
+            b = by.dot(x).conjugate()
+
+            # similar to numpy all close, but normalised with respect to operator, x and y
+            error = numpy.abs( a - b )/ (operator.norm()*x.norm()*y.norm() + 1e-12)
+
+            if error < tol:
+                return True
+            else:
+                print ('Left hand side  {}, \nRight hand side {}'.format(a, b))
+                return False    

```

Comment:
I agree, better to use something like
`self.assert_less(error,tol)`

You can add an error message to the assert if you want more information, as standard it'll print the error and tol however.

Reply:
If we decide for this, this will change the current behaviour of the `dot_test` which will either return True or raise an Error. Are we ready for that?